nix/build-support/stacklock2nix/default.nix

```diff
@@ -513,7 +513,7 @@ let
     let
       # The localPkgPathStr is assumed to be a relative path from the
       # directory containing the `stack.yaml` file.
-      rawPkgPath = builtins.path { path = dirOf stackYamlReal + ("/" + localPkgPathStr); };
+      rawPkgPath = dirOf stackYamlReal + ("/" + localPkgPathStr);
```

Comment:
The main problem stems from here.

If you scroll down, you can see that `rawPkgPath` is being passed to `lib.cleanSourceWith` on line 607 (and one other place).

However, `lib.cleanSourceWith` internally passes the `src` argument to `builtins.path`.  And calls to `builtins.path` don't compose correctly (so the parent paths end up in the Nix store).  This is why `lib.cleanSourceWith` is needed in the first place.  The value passed to the `src` argument of `lib.cleanSourceWith` should _not_ be defined with `builtins.path` (in order to correctly use the composition functionality of `lib.cleanSourceWith`).

This PR changes the definition of `rawPkgPath` to not use `builtins.path`, and instead correctly rely on `lib.cleanSourceWith` to do it for us.